service/src/test/java/bio/terra/profile/common/MetricsUtilsUnitTest.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.profile.common;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import bio.terra.profile.app.common.MetricUtils;
+import bio.terra.profile.model.CloudPlatform;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class MetricsUtilsUnitTest extends BaseUnitTest {
+
+  private SimpleMeterRegistry meterRegistry;
+
+  @BeforeEach
+  void setUp() {
+    meterRegistry = new SimpleMeterRegistry();
+    Metrics.globalRegistry.add(meterRegistry);
+  }
+
+  @AfterEach
+  void tearDown() {
+    meterRegistry.clear();
+    Metrics.globalRegistry.clear();
+  }
+
+  @Test
+  void createGcpProfileMetrics() {
+    var profile = ProfileFixtures.createGcpBillingProfile("ABCD1234");
+
+    MetricUtils.recordProfileCreation(() -> profile, profile.cloudPlatform());
```

Comment:
By doing the test separately, we have no coverage that we actually call `recordProfileCreation` as part of profile creation. Can you spy on that call in `CreateProfileFlightTest` to also ensure the method is called?